clients/src/main/java/org/apache/kafka/clients/consumer/internals/Coordinator.java

```diff
@@ -460,6 +460,12 @@ private void maybeAutoCommitOffsetsSync() {
                     offsetAndMetadata.offset(), offsetAndMetadata.metadata()));
         }
 
+        // reset generation/consumerId to default values if using manual assignment
+        if (!subscriptions.partitionsAutoAssigned()) {
+            this.consumerId = OffsetCommitRequest.DEFAULT_CONSUMER_ID;
```

Comment:
For defaults: agree, we can file a separate ticket for moving all the default values to Protocol.